consensus_test.go

```diff
@@ -11,89 +15,24 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/trace"
 )
 
-func TestTransition_ValidateState_Prepare(t *testing.T) {
-	t.Skip()
-
-	// we receive enough prepare messages to lock and commit the proposal
-	i := newMockPbft(t, []string{"A", "B", "C", "D"}, "A")
-	i.setState(ValidateState)
-
-	i.emitMsg(&MessageReq{
-		From: "A",
-		Type: MessageReq_Prepare,
-		View: ViewMsg(1, 0),
-	})
-	i.emitMsg(&MessageReq{
-		From: "B",
-		Type: MessageReq_Prepare,
-		View: ViewMsg(1, 0),
-	})
-	// repeated message is not included
-	i.emitMsg(&MessageReq{
-		From: "B",
-		Type: MessageReq_Prepare,
-		View: ViewMsg(1, 0),
-	})
-	i.emitMsg(&MessageReq{
-		From: "C",
-		Type: MessageReq_Prepare,
-		View: ViewMsg(1, 0),
-	})
-
-	i.runCycle(context.Background())
+var (
+	mockProposal  = []byte{0x1, 0x2, 0x3}
+	mockProposal1 = []byte{0x1, 0x2, 0x3, 0x4}
+)
 
-	i.expect(expectResult{
-		sequence:    1,
-		state:       ValidateState,
-		prepareMsgs: 3,
-		commitMsgs:  1, // A commit message
-		locked:      true,
-		outgoing:    1, // A commit message
-	})
+// MockedTracer is a mocked object that implements Tracer interface
+type MockedTracer struct {
```

Comment:
I think we do not need this right? It does not panic AFAIK when it is not set.

Reply:
Indeed, will remove it.